packages/react/src/primitives/types/passwordField.ts

```diff
@@ -61,7 +62,9 @@ export interface PasswordFieldProps extends TextFieldProps {
 
 export type PasswordFieldType = 'password' | 'text';
 
-export interface ShowPasswordButtonProps extends ButtonProps {
+export interface ShowPasswordButtonProps
+  extends ButtonProps,
+    Pick<InputProps, 'hasError'> {
```

Comment:
To resolve the conflicts, this should be 
```suggestion
  extends BaseButtonProps,
    Pick<BaseInputProps, 'hasError'> {
```